### ISSUE TITLE
A project overview and Tracker

### ISSUE THREAD

**Issue by derickcaps** [coleam00/bolt.new-any-llm#120]
Is there a way we could add an agent that works in the background and constantly updates the app step by step instructions with specific details as well as being able to open and edit this automatically updated prompt window. 

For example when I start to put alot of time and things into complex tasks the fine details sometimes get lost like a specific Identifier and what to do as a next step eventually gets lost and if I don't explicity say it, I could be doing circles but sometimes if I ask for a step by step processes I can see where the error is because they misinterpreted something and are now doing something incorrectly over and over.

So basically a planner that keeps tracks of changes and features without wrecking the prompt or task overview so you can see what it thinks the job is and can make it so you don't lose logic at max context length

**Comment by mikl0s** [comment]
Pythagoras works sort of like that - the architect AI creates epics with 4-5 stories per epic from the initial prompt, then you progress through and the user approve each story. The downside of their way is that the first tasks does not have proper context of later things so it will redo stuff over and over.

An improved way of doing this would be the absolute killer feature.
